### PR TITLE
ref(onboarding): Improve 'getPlatformPath' func

### PR DESCRIPTION
**Pull request by priscilawebdev**
### TL;DR
This PR improves the readability of the [getPlatformPath](https://github.com/getsentry/sentry/blob/a5e9063aa4e2d84fc6baad75e1fcc4d015a3ce92/static/app/utils/gettingStartedDocs/getPlatformPath.ts#L3) function. Consequently, updates were made to files that use this function.

### Reason
During the migration of the Onboarding Docs, we created the getPlatformPath function. This function converts a platform ID, type, and language into a path matching the Sentry repository's documentation structure.

Initially, I considered removing this function. However, a similar function is used in [this file](https://github.com/getsentry/sentry/blob/a5e9063aa4e2d84fc6baad75e1fcc4d015a3ce92/static/app/components/feedback/feedbackOnboarding/useLoadFeedbackOnboardingDoc.tsx#L9), suggesting that these functions should be merged in a follow-up PR.

This PR, at minimum, cleans up the function, making it more readable.

### Consequences

- A few files in the Getting Started Docs structure and test files have been updated. Now they are aligned with the structure's logic:
  -  language/language
  -  language/framework

- [A new test file](https://github.com/getsentry/sentry/blob/a3a84041aca0a7c4b0caf5cfbca7d24995fc485d/static/app/components/onboarding/gettingStartedDoc/sdkDocumentation.spec.tsx) was created to ensure everything runs smoothly.

### Disclaimer

Each platform was tested manually and everything is working as it should.

### CODE REVIEW

**Review by vgrozdanic — APPROVED** [review]

File: static/app/utils/gettingStartedDocs/getPlatformPath.spec.ts [hunk 1]
```diff
@@ -49,34 +49,41 @@ describe('getPlatformPath', () => {
 
     // Frameworks
     expect(getPlatformPath(getFrameworkPlatformWithId('capacitor'))).toEqual(
-      'capacitor/capacitor'
+      'language/language'
+    );
+
+    expect(getPlatformPath(getFrameworkPlatformWithId('ionic'))).toEqual(
+      'language/language'
+    );
```
Comment (ArthurKnaus) [inline static/app/utils/gettingStartedDocs/getPlatformPath.spec.ts:57]: Not sure how useful the "handles special cases" test is anymore 🤔 
imho we should substitute that with a test for the `platform.id === platform.language` logic branch.

**Review by ArthurKnaus — APPROVED** [review]
great work!
